tools/syz-db/syz-db.go

```diff
@@ -221,3 +237,54 @@ func bench(target *prog.Target, file string) {
 	fmt.Printf("program size: min=%v avg=%v max=%v 10%%=%v 50%%=%v 90%%=%v\n",
 		lens[0], sum/n, lens[n-1], lens[n/10], lens[n/2], lens[n*9/10])
 }
+
+func print(file string) {
+	db, err := db.Open(file, false)
+	if err != nil {
+		tool.Failf("failed to open database: %v", err)
+	}
+	var keys []string
+	for key := range db.Records {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		rec := db.Records[key]
+		fmt.Printf("%v\n%v\n", key, string(rec.Val))
+	}
+}
+
+func rm(file, syscall string) {
+	db, err := db.Open(file, false)
+	if err != nil {
+		tool.Failf("failed to open database: %v", err)
+	}
+	for key, rec := range db.Records {
+		result := removeLinesContaining(rec.Val, syscall)
+		if len(result) > 0 {
+			db.Save(key, result, rec.Seq)
+		} else {
+			delete(db.Records, key)
+		}
+	}
+	if err := db.Flush(); err != nil {
+		tool.Failf("failed to flush db")
+	}
+}
+
+func removeLinesContaining(input []byte, target string) []byte {
+	var output bytes.Buffer
+	reader := bufio.NewReader(bytes.NewReader(input))
+
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			break
+		}
+		if !bytes.Contains([]byte(line), []byte(target)) {
+			output.WriteString(line)
```

Comment:
If we just remove lines, it may leave inconsistent programs with resources that weren't created.
E.g. if we remove open from:
```
r0 = open(...)
close(r0)
```
we will get:
```
close(r0)
```

How does manager handle such inconsistent programs?

Reply:
I added a test case and tried to use p.validate which doesn't return any error.
Any better way to verify such case?